tests/sentry/search/events/builder/test_metrics.py

```diff
@@ -2501,6 +2504,43 @@ def test_run_top_timeseries_query_with_on_demand_columns(self):
             ],
         )
 
+    def test_run_top_timeseries_query_with_unicode_string(self) -> None:
+        for feature, expected_version, expected_flags in [
+            ("", 1, set()),
+            (FEAT_FLAG, 2, {"use_new_unicode_and_env_logic"}),
+        ]:
+            with Feature(feature):
+                unicode_str = "Hello, 世界!"
+                field = "count()"
+                groupbys = ["customtag1"]
+                query_s = "transaction.duration:>=100"
+                spec = OnDemandMetricSpec(field=field, groupbys=groupbys, query=query_s)
+
+                for day in range(0, 5):
+                    self.store_on_demand_metric(
+                        day * 62 * 24,
+                        spec=spec,
+                        additional_tags={"customtag1": unicode_str},
+                        timestamp=self.start + datetime.timedelta(days=day),
+                    )
+
+                query = TopMetricsQueryBuilder(
+                    Dataset.PerformanceMetrics,
+                    self.params,
+                    3600 * 24,
+                    [{"customtag1": unicode_str}],
+                    query=query_s,
+                    selected_columns=groupbys,
+                    timeseries_columns=[field],
+                    config=QueryBuilderConfig(
+                        on_demand_metrics_enabled=True,
+                        on_demand_metrics_type=MetricSpecType.SIMPLE_QUERY,
+                    ),
+                )
+                spec_version = query._on_demand_metric_spec_map["count()"].spec_version
+                assert spec_version.version == expected_version
+                assert spec_version.flags == expected_flags
```

Comment:
These two assertions prove that the right spec version is picked up depending on the features enabled for an org.